### PR TITLE
SceneLoader Options

### PR DESCRIPTION
**Pull request by ryantrem**
This is an alternative approach to https://github.com/BabylonJS/Babylon.js/pull/15222. Compared to the original approach, this one:
1. Has a simpler implementation and has a smaller impact on the API surface.
1. Is slightly less type safe when adding new loader plugins or glTF extensions.
1. Is much simpler for the caller.

With this approach, options are a type safe plain object with no factory functions. Invocation looks like this (for example):
```ts
await SceneLoader.LoadAssetContainerAsync("https://raw.githubusercontent.com/BabylonJS/Assets/master/meshes/LevelOfDetail.glb", scene, {
  onProgress: (event) => { },
  pluginOptions: {
    gltf: {
      loadOnlyMaterials: true,
      extensionOptions: {
        MSFT_lod: {
          maxLODsToLoad: 5,
        },
        MSFT_minecraftMesh: {
          enabled: false,
        },
      },
    },
  },
});
```

This works by simply allowing plugins and extensions to augment the options interface, such that when you import a plugin or extension, it augments the options interface with options specific to the plugin or extension. From the consumer's standpoint, this is fully type safe, so you get clear intellisense showing you exactly how the options can be configured.

I only implemented options for glTF in this PR, not any other loaders.

This is how it shows up with intellisense in VS Code:
image
image

@deltakosh, @sebavan, @bghgary

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15344/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15344/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15344/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15344/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15344/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15344/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15344/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15344/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15344/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15344/merge/testResults/webgpuplaywright/index.html

**Comment by deltakosh** [comment]
I like that version more! This is far better from a user standpoint

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15344/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15344/merge/testResults/webgpuplaywright/index.html

**Review by sebavan — APPROVED** [review]

**Comment by sebavan** [comment]
Would be cool to add a test or two using the new options

**Comment by ryantrem** [comment]
> Would be cool to add a test or two using the new options

Agreed, it's on my todo list. I could include it with this PR or create a separate PR, either way is fine with me.

**Comment by sebavan** [comment]
either way is fine with me as well

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Loading/sceneLoader.ts [hunk 1]
```diff
@@ -369,6 +370,114 @@ interface IFileInfo {
     rawData: Nullable<ArrayBufferView>;
 }
 
+/**
+ * Defines options for SceneLoader plugins. This interface is extended by specific plugins.
+ */
+export interface SceneLoaderPluginOptions extends Record<string, Record<string, unknown> | undefined> {}
```
Comment (deltakosh) [inline packages/dev/core/src/Loading/sceneLoader.ts:376]: Maybe on a separate file?

**Review by bghgary — APPROVED** [review]

File: packages/dev/core/src/Loading/sceneLoader.ts (hunk 1 above)
Comment (ryantrem) [reply packages/dev/core/src/Loading/sceneLoader.ts:376]: Do you mean just this one interface, or all the interfaces I added that are options related? My preference would be to leave them here as it seems simpler for the user (they are just importing SceneLoader related types from a single file). If you feel strongly that this one (and/or others) should be moved to another file, will you share a little more of your thinking?

Comment (ryantrem) [reply packages/dev/core/src/Loading/sceneLoader.ts:376]: Chatted offline, we can follow up on this for a later PR if you'd like!

**Pull request merged**